File: KotlinAlter/src/main/kotlin/com/company/cntrl/key/SideFactory.kt
package com.company.cntrl.key

object Side {
    val TOP: Int = 38
    val BOTTOM: Int = 40
    val LEFT: Int = 37
    val RIGHT: Int = 39
}


object SideFactory {
    fun getMoveSide(key: Int): Repainter {
        return when (key){
            Side.TOP -> UpMove()
            Side.BOTTOM -> DownMove()
            Side.LEFT -> LeftMove()
            Side.RIGHT -> RightMove()
            else -> UpMove()
        }
    }

}

File: ScalaAlt/src/main/scala/cntrl/keys/SideFactory.scala
package cntrl.keys

object Side extends Enumeration {
  val TOP = 38
  val BOTTOM = 40
  val LEFT = 37
  val RIGHT = 39

}

object SideFactory {

  def getMoveSide(key: Int): BallRepainter = key match {
    case Side.TOP => new UpMove()
    case Side.BOTTOM => new DownMove()
    case Side.LEFT => new LeftMove()
    case Side.RIGHT => new RightMove()
  }

}
